activesupport/lib/active_support/evented_file_update_checker.rb

```diff
@@ -55,17 +55,15 @@ def initialize(files, dirs = {}, &block)
       dtw = directories_to_watch
       @dtw, @missing = dtw.partition(&:exist?)
 
-      if @dtw.any?
```

Comment:
Closer to the original implementation might be `@dtw.any? || @missing.any?`, but I do prefer this new behaviour of just always requiring it. 👍

Reply:
In fact, this allows to defer the booting to `#updated?` if there is no `@dtw`, which call `#boot!` only when a `@missing`  folder becomes a `@dtw` (and return `true`)